packages/core/src/__tests__/dispatcher-mock-ipfs.test.ts

```diff
@@ -253,4 +256,67 @@ describe('Dispatcher with mock ipfs', () => {
     await dispatcher.handleMessage({ typ: MsgType.RESPONSE, id: queryID, tips: tips })
     expect(dispatcher.repository.stateManager.update).toBeCalledWith(stream2.id, FAKE_CID2)
   })
+
+  it('handle message correctly with model', async () => {
+    dispatcher.repository.stateManager = {} as unknown as StateManager
+
+    async function register(state: StreamState) {
+      const runningState = new RunningState(state, false)
+      repository.add(runningState)
+      dispatcher.messageBus.queryNetwork(runningState.id).subscribe()
+      return runningState
+    }
+
+    const initialState = {
+      type: 0,
+      log: [
+        {
+          cid: FAKE_STREAM_ID.cid,
+          type: CommitType.GENESIS,
+        },
+      ],
+    } as unknown as StreamState
+    const state$ = await register(initialState)
+
+    // Store the query ID sent when the stream is registered so we can use it as the response ID later
```

Comment:
no need for testing QUERY or RESPONSE messages in this test, they are already tested in the previous test you copied this one from, that would just be redundant duplicate testing.  I think this test case can be made way smaller and only focused on the UPDATE message